nxontology_ml/model/predict.py

```diff
@@ -0,0 +1,115 @@
+from nxontology_ml.model.train import train_model
+
+NON_DISEASE_THERAPEUTIC_AREAS: set[str] = {
+    "EFO:0005932",  # animal_disease
+    "EFO:0001444",  # measurement
+    "EFO:0000651",  # phenotype
+    "GO:0008150",  # biological_process
+    "EFO:0002571",  # medical_procedure
+}
+
+
+def get_disease_nodes(
+    take: int | None = None,
+    nxo: NXOntology[str] | None = None,
+) -> Iterable[str]:
+    """
+    If all roots are in the set of non-disease therapeutic areas (aka roots in EFO OTAR Slim),
+        then its non-disease
+    """
+    nxo = nxo or get_efo_otar_slim()
+    for n in islice(sorted(nxo.graph), take):
+        if any(
+            root not in NON_DISEASE_THERAPEUTIC_AREAS for root in nxo.node_info(n).roots
+        ):
+            yield n
+
+
+rs_cls_to_precision = {
+    "01-disease-subtype": "high",
+    "02-disease-root": "medium",
+    "03-disease-area": "low",
+}
+
+
+def predict(
+    feature_pipeline: Pipeline,
+    model: CatBoostClassifier,
+    training_set: tuple[np.ndarray, np.ndarray],
+    nxo: NXOntology[str],
+    include_feature_values: bool = True,
+    take: int | None = 0,
+) -> pd.DataFrame:
+    target_nodes: list[str] = list(get_disease_nodes(take=take, nxo=nxo))
+    target_features = feature_pipeline.transform(target_nodes)
+    target_labels = model.predict(target_features)
+    target_probas = model.predict_proba(target_features)
+    training_set_labels = {
+        id_: str(label) for id_, label in zip(*training_set, strict=True)
+    }
+    model_df = pd.DataFrame(
+        [
+            {
+                "identifier": node_id,
+                "precision": rs_cls_to_precision[label[0]],
+                "proba_high": probas[0],
+                "proba_medium": probas[1],
+                "proba_low": probas[2],
+                "efo_label": nxo.node_info(node_id).data["efo_label"],
+                "rs_classification": training_set_labels.get(node_id, None),
+            }
+            for node_id, label, probas in zip(
+                target_nodes, target_labels, target_probas, strict=True
+            )
+        ]
+    )
+    if include_feature_values:
+        features_output_df = pd.DataFrame(
+            data=np.hstack(
+                [
+                    target_features.cat_feature_data,
+                    np.array(target_features.num_feature_data, dtype=float),
+                ]
+            ),
+            columns=np.array(
+                target_features.cat_feature_names + target_features.num_feature_names
+            ),
+        )
+        model_df = pd.concat([model_df, features_output_df], axis=1).convert_dtypes()
+    return model_df
+
+
+def train_predict(
+    conf: ModelConfig = DEFAULT_MODEL_CONFIG,
+    nxo: NXOntology[str] | None = None,
+    training_set: tuple[np.ndarray, np.ndarray] | None = None,
+    include_feature_values: bool = True,
+    train_take: int | None = 0,
+    predict_take: int | None = 0,
+) -> pd.DataFrame:
```

Comment:
Nice! This looks like the right arguments to expose.

Perhaps we add a docstring here with parameter descriptions. Specifically, not immediately sure what `train_take` and `predict_take` are doing.

Reply:
> Perhaps we add a docstring here with parameter descriptions. Specifically, not immediately sure what train_take and predict_take are doing.

That's a good point: `train_take` and `predict_take` allow to take a subset of the points for training and/or prediction to speed up the whole process (mostly useful when manually or automatically testing).

Will add the doc.